File: Backend/src/main/java/com/project/consphere/service/FollowService.java
package com.project.consphere.service;

import com.project.consphere.dto.FollowResponse;

import java.util.List;

public interface FollowService {

    void followUser(String username);

    void unfollowUser(String username);

    List<FollowResponse> getFollowers();

    List<FollowResponse> getFollowing();
}

File: frontend/src/services/followService.js
import api from './api';

export const followService = {
  followUser: async (username) => {
    const response = await api.post(`/follow/${username}`);
    return response.data;
  },

  unfollowUser: async (username) => {
    const response = await api.delete(`/follow/${username}`);
    return response.data;
  },

  getFollowers: async () => {
    const response = await api.get('/follow/followers');
    return response.data;
  },

  getFollowing: async () => {
    const response = await api.get('/follow/following');
    return response.data;
  }
};
